compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -684,21 +686,20 @@ fn quoted_as_expr(
 ) -> IResult<Value> {
     let argument = check_one_argument(arguments, location)?;
 
-    let result =
-        parse(interner, argument.clone(), Parser::parse_expression_or_error, "an expression");
-    if let Ok(expr) = result {
-        return option(return_type, Some(Value::expression(expr.kind)));
-    }
-
-    let result =
-        parse(interner, argument.clone(), Parser::parse_statement_or_error, "an expression");
-    if let Ok(stmt) = result {
-        return option(return_type, Some(Value::statement(stmt.kind)));
-    }
+    let result = parse(
+        interner,
+        argument.clone(),
+        Parser::parse_statement_or_expression_or_lvalue,
+        "an expression",
+    );
 
-    let result = parse(interner, argument, Parser::parse_lvalue_or_error, "an expression");
-    if let Ok(lvalue) = result {
-        return option(return_type, Some(Value::lvalue(lvalue)));
+    if let Ok(statement_or_expression_or_lvalue) = result {
+        let value = match statement_or_expression_or_lvalue {
+            StatementOrExpressionOrLValue::Expression(expr) => Value::expression(expr.kind),
+            StatementOrExpressionOrLValue::Statement(statement) => Value::statement(statement.kind),
+            StatementOrExpressionOrLValue::LValue(lvalue) => Value::lvalue(lvalue),
+        };
+        return option(return_type, Some(value));
```

Comment:
Could use `result.ok().map(...)` so that we don't need to early-return and only need the final call to `option`

Reply:
Good idea, done!